### ISSUE TITLE
A around_step hook would be really helpful

### ISSUE THREAD

**Issue by Jack12816** [codegram/spinach#201]
Hey guys :) first things first: thanks for this great gem!

I really enjoy the flexibility which is offered by all these hooks. But the other day I recognized there is no `around_step` hook. My issue was related to a SPA which was built with React. One of my test steps evaluated the text content of a node.  React Component fills in "Hello undefined!" due to the missing forename of the user.  A few milliseconds later the API call finished and the state gets refreshed with the correct data. But then Capybara already found the node and Rspec failed. So even with the help of spinach-rerun-reported it is not stable.

So I need something like this:
```ruby
require 'retries'
require 'rspec/expectations'

# Retry a single step multiple times before failing.
# This takes only care of rspec expectations.
Spinach.hooks.around_step do |step_data, step_definitions, &block|
  exceptions = [RSpec::Expectations::ExpectationNotMetError]
  with_retries(max_tries: 5, rescue: exceptions) { block.call }
end
```

PR is on the way :)

**Comment by Jack12816** [comment]
This is really handy in situations like this:
```ruby
step 'I see a greeting' do
  expect(page.find('.welcome span span').text).to \
    be_eql 'Willkommen, Max Mustermann.'
end
```
```
    ✘  And I see a greeting                                        # features/steps/login.rb:34
        expected `"Willkommen, undefined undefined.".eql?("Willkommen, Max Mustermann.")` to return true, got false
```

**Comment by ywen** [comment]
If you use `capyabara` then you can write it as:

```ruby
expect(page).to have_text("Willkommen, Max Mustermann.")
```

If you use this, capybara will wait for a given time and keep retrying for you until a preset timeout is reached. At least in this case you don't need an `around_step`

**Comment by Jack12816** [comment]
Thank you - this is great. :) It also works in a element specific way:
```ruby
expect(page.find('.welcome span span')).to \
  have_text('Willkommen, Max Mustermann.') 
```

But nevertheless, such a hook could be handy in other use cases or to support the first problem.

**Comment by oriolgual** [comment]
Closed by #202
